config/src/generator.rs

```diff
@@ -75,12 +77,14 @@ pub fn validator_swarm(
         // TODO(philiphayes): as a temporary hack, we'll just duplicate the
         // validator info into the fullnode info until we can handle deserializing
         // empty fullnode info.
+        let addr = NetworkAddress::try_from(network.advertised_address.clone()).unwrap();
+        let addr = RawNetworkAddress::try_from(&addr).unwrap();
```

Comment:
Isn't `network.advertised_address` a NetworkAddress already? Why do we need to convert it?

Also, shouldn't we be avoiding use of unwrap()?